src/mcmc/algorithm.jl

```diff
@@ -11,4 +12,35 @@ this wrapping occurs automatically.
 """
 abstract type InferenceAlgorithm end
 
+# TODO(penelopeysm): remove
 DynamicPPL.default_chain_type(sampler::Sampler{<:InferenceAlgorithm}) = MCMCChains.Chains
+
+"""
+    update_sample_kwargs(spl::AbstractSampler, N::Integer, kwargs)
+
+Some samplers carry additional information about the keyword arguments that
+should be passed to `AbstractMCMC.sample`. This function provides a hook for
+them to update the default keyword arguments. The default implementation is for
+no changes to be made to `kwargs`.
+"""
+update_sample_kwargs(::AbstractSampler, N::Integer, kwargs) = kwargs
+
+"""
+    get_adtype(spl::AbstractSampler)
+
+Return the automatic differentiation (AD) backend to use for the sampler. This
+is needed for constructing a LogDensityFunction. By default, returns nothing,
+i.e. the LogDensityFunction that is constructed will not know how to calculate
+its gradients. If the sampler requires gradient information, then this function
+must return an `ADTypes.AbstractADType`.
+"""
+get_adtype(::AbstractSampler) = nothing
+
+"""
+    requires_unconstrained_space(sampler::AbstractSampler)
```

Comment:
Maybe this is a unimportant point: for something like MH, both constrained and unconstrained spaces might be fine.

On a high level, I do think "**require**s_unconstrained_space"-ness is a sampler property, but it's not sufficient condition to control linking.

Don't know if this makes sense.